opentracing/tracer.py

```diff
@@ -37,16 +39,71 @@ class Tracer(object):
 
     _supported_formats = [Format.TEXT_MAP, Format.BINARY, Format.HTTP_HEADERS]
 
-    def __init__(self):
+    def __init__(self, scope_manager=ScopeManager()):
```

Comment:
I think we should have `scope_manager=None` and then initialize it if not set. If we use `ScopeManager()` as default value the object will be shared across multiple calls. Here a code example of the result that we may don't want:

```python
class ScopeManager(object):
    def __init__(self):
        self.a = 0

    def increment(self):
        self.a += 1
        return self.a


def init(scope_manager=ScopeManager()):
    return scope_manager.increment()


print(init())
print(init())

# output:
# 1
# 2 <-- it should be '1' since we may expect it as a new ScopeManager instance
```

Reply:
Will update :) Even though its supposed to be a 'noop' `ScopeManager`, I'd rather be on the safe side, as you suggest ;)